### ISSUE TITLE
Resource checks can wait for multiple checker intervals before being run

### ISSUE THREAD

**Issue by clarafu** [concourse/concourse#5730]
## Summary

Occasionally we see that resources can be slow to get checked and usually these resources are ones are sharing resource configs with other resources. For example, I created a local deployment with docker compose and I set 10 pipelines using the [`lidar-test` pipeline](https://github.com/concourse/drills/blob/master/longevity/lidar-test/pipelines/passed-jobs.yml) in the drills repo but I added the unique param to the `concourse-docker-git` and the `concourse-docs-git` resources. I wanted these resources to have their own check containers because I wanted the `concourse` resource to be the only resource that shares a check container between multiple resources. Since those resources use the same resource type as the `concourse` resource, the only difference between them would be the fact that the `concourse` resources share a check container.

When all the pipelines were running, I greped for the log line `check-already-tracked` which is outputted when the check exits because it was unable to acquire the lock for it's resource config.

https://github.com/concourse/concourse/blob/153a8c4ff4bb8609396289650972127e84592c27/atc/engine/engine.go#L360-L363

I saw that there were a lot of log lines with `check-already-tracked`, basically one every second. 

![Screen Shot 2020-06-09 at 12 21 19 PM](https://user-images.githubusercontent.com/19290703/84174005-079a2380-aa4c-11ea-9788-4cc6f485ff20.png)

And when I queried the database for any resources that have not been checked within 2 minutes (which is already 1 minute more than the resource checking interval of 1 minute), I saw that the only resources that have not been checked within 2 minutes were the `concourse` resources.

![Screen Shot 2020-06-09 at 12 20 47 PM](https://user-images.githubusercontent.com/19290703/84174028-0ec13180-aa4c-11ea-814c-70f1fd92abb9.png)

I think with us sharing the same check container between resources that share a resource config, that will cause lock contention for that check container but I think the slowness of the checks is exacerbated by the fact that we fail out of the check if the lock is already acquired. This means that we will only try to run that check again in the next interval of the checker. If we have 10 resources that are trying to check the same resource config and the checker runs every 10 seconds, it can be up to 100 seconds until we can run the check for a resource.

Maybe a temporary solution, before we fix this in the long-term by having check containers be per resource config scope rather than resource config, can be to wait on the resource config lock rather than returning out? This will make it so that we don't need to wait for the next interval of the checker to pick up the check.

## Triaging info

* Concourse version: 6.2
* Did this used to work? Nope

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
